src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,111 @@
+                    throw new DukeException("Please use yyyy-mm-dd as the date format.");
+                }
+            case "event":
+                if (str.length != 2) {
+                    throw new DukeException("The description of event cannot be empty.");
+                }
+                splitDescription = str[1].split(" /at ", 2);
+                if (splitDescription.length != 2) {
+                    throw new DukeException("Command are missing either description or remark.");
+                }
+                return new EventCommand(splitDescription[0], splitDescription[1]);
+            case "mark":
+                try {
+                    int idx = Integer.parseInt(str[1]);
+                    try {
+                        return new MarkCommand(idx - 1);
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("Invalid index.");
+                    }
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("Argument of mark cannot be empty.");
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The argument you've input is not an integer");
+                }
+            case "unmark":
+                try {
+                    int idx = Integer.parseInt(str[1]);
+                    try {
+                        return new UnmarkCommand(idx - 1);
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("Invalid index.");
+                    }
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("Argument of unmark cannot be empty.");
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The argument you've input is not an integer");
+                }
+            case "delete":
+                try {
+                    int idx = Integer.parseInt(str[1]);
+                    try {
+                        return new DeleteCommand(idx - 1);
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("Invalid index.");
+                    }
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("Argument of delete cannot be empty.");
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The argument you've input is not an integer");
+                }
+            case "find":
+                try {
+                    if (str[1].length() == 0) {
+                        throw new DukeException("The description of find cannot be empty.");
+                    }
+                    return new FindCommand(str[1]);
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("The description of find cannot be empty.");
+                }
```

Comment:
Quite a bit of nesting here, maybe you could consider abstracting the code in each case into their relevant methods?